### ISSUE TITLE
Spatial viewer not functioning with SRID 2232

### ISSUE THREAD

**Issue by rustprooflabs** [dbeaver/dbeaver#8553]
#### System information: 
- Operating system (distribution) and version: Windows 10 Pro
- DBeaver version: 7.0.3 (first noticed on 7.0.2)
- Additional extensions: N/A

#### Connection specification:
- Database name and version: Pg12, PostGIS 3.

```sql
SELECT version(), postgis_version();
```

```
Name           |Value                                                                                                   |
---------------|--------------------------------------------------------------------------------------------------------|
version        |PostgreSQL 12.2 on armv7l-unknown-linux-gnueabihf, compiled by gcc (Raspbian 8.3.0-6+rpi1) 8.3.0, 32-bit|
postgis_version|3.0 USE_GEOS=1 USE_PROJ=1 USE_STATS=1                                                                   |
```

- Driver name
Postgres JDBC

- Do you use tunnels or proxies (SSH, SOCKS, etc)? 
No

#### Describe the problem you're observing:

Points in SRID 2232 do not render the point or the base map.  If the base map does not render that would be OK, but all I am presented with is a gray screen in the spatial tab, my points do not render at all.  This occurs in both the spatial tab and the spatial display in the Value panel.

I first encountered the issue with 7.0.2.  Upgraded to 7.0.3 and am still experiencing the issue.

#### Steps to reproduce, if exist:

```sql
-- Displays point over basemap
SELECT ST_SetSRID(ST_Point(-105, 40), 4326);
-- Displays point over basemap
SELECT ST_Transform(ST_SetSRID(ST_Point(-105, 40), 4326), 2773);
-- No point, no basemap
SELECT ST_Transform(ST_SetSRID(ST_Point(-105, 40), 4326), 2232);
```

Screenshots showing results of each of the three queries.  First.

![image](https://user-images.githubusercontent.com/3085224/80286160-08b70180-86e7-11ea-8ef3-e9813d3a98f3.png)

Second, also works.

![image](https://user-images.githubusercontent.com/3085224/80286173-1b313b00-86e7-11ea-89c6-0290df066a5c.png)

Just a blank gray display.

![image](https://user-images.githubusercontent.com/3085224/80286190-2dab7480-86e7-11ea-9d3a-b356b8679eae.png)

**Comment by uslss** [comment]
thanks for the bug report

**Comment by serge-rider** [comment]
We use [CTS](https://github.com/orbisgis/cts) library for geometry transformations.

It doesn't support 2232->4326 (GPS) transformations. Unfortunately we can't fix this from DBeaver side.
CTS wasn't updated for more than a year. Maybe we should switch to some better alternative but I'm not sure that it exists..

Note: we can't use PostGIS for that as geometry transforms works on client side for different databases, not just PostgreSQL.

**Comment by rustprooflabs** [comment]
Thanks for trying!

Is that limitation documented somewhere in CTS or found it in the code?  It'd be great to have a cheat-sheet of SRIDs that don't work.  Or a full list of those that do work.  I was pleasantly surprised when I discovered 2773 (same region, just in meters instead of feet) worked instead of having to transform to 3857 or 4326 manually.

**Comment by serge-rider** [comment]
I don't think that there is such table anywhere. There are thousands of SRIDs, it would be a table of millions of rows.
Anyway, transforms don't work like this. There are algorithms which do conversions based on CRS definition. In most cases there are series of consequent transformations.

I can only assume that most of feet-based transforms won't work :(

**Comment by HughKelley** [comment]
I've run into this issue as well. So far, it seems that in North America, CTS may only handle HARN coordinate reference systems and older systems are more likely to fail. 
That's consistent with the example above comparing 2773 (a HARN crs) to 2232 (which predates HARN)

[https://www.georepository.com/crs_4152/NAD83-HARN.html](https://www.georepository.com/crs_4152/NAD83-HARN.html)

**Comment by heidivanparys** [comment]
> We use [CTS](https://github.com/orbisgis/cts) library for geometry transformations.
> 
> It doesn't support 2232->4326 (GPS) transformations. Unfortunately we can't fix this from DBeaver side. CTS wasn't updated for more than a year. Maybe we should switch to some better alternative but I'm not sure that it exists.

PROJ could be used instead? See

- https://proj.org/
- https://github.com/OSGeo/PROJ
- https://github.com/OSGeo/PROJ-JNI

(I'm not involved in PROJ myself, but I'm sure the people behind those repositories can provide more info or answer questions.)

**Comment by mkgrgis** [comment]
Spatial viewer's available web map services should also depends on SRID, 2232 is ♁ Earth SRID, but 104904 and 104905 is ♂ Mars SRID. Earth WMS like Google or OSM or ESRI satellite in case of geodata about Mars looks not normal. https://mkgrgis.github.io/openPlanetWMS/♂.html shows possible data sources for Mars. @serge-rider , should I register this problem as separate issue? I can help with Leaflet `tileLayers` WMS URL string for Mars for some SRIDs.
